src/auth/jwtaccess.ts

```diff
@@ -57,26 +57,38 @@ export class JWTAccess {
   /**
    * Get a non-expired access token, after refreshing if necessary.
    *
-   * @param {string} authURI the URI being authorized
+   * @param authURI The URI being authorized.
+   * @param additionalClaims An object with a set of additional claims to
+   * include in the payload.
    * @returns An object that includes the authorization header.
    */
-  getRequestMetadata(authURI: string): RequestMetadataResponse {
+  getRequestMetadata(
+      authURI: string,
+      additionalClaims?: {[index: string]: string}): RequestMetadataResponse {
     const cachedToken = this.cache.get(authURI);
     if (cachedToken) {
       return cachedToken;
     }
     const iat = Math.floor(new Date().getTime() / 1000);
     const exp = iat + 3600;  // 3600 seconds = 1 hour
 
+    // if additionalClaims are provided, ensure they do not collide with
+    // other required claims.
+    if (additionalClaims) {
+      ['iss', 'sub', 'aud', 'exp', 'iat'].forEach(claim => {
```

Comment:
nit: If a new property is added to the inline object at line 90, someone will need to remember to come back to this list and update it. Instead move the object at line 90 to a named object above, and then iterate the keys of the object to make sure none of them are in additional claims.

Reply:
That is awesome feedback.